src/TriggerBinding/SqlTriggerTargetScaler.cs

```diff
@@ -14,24 +17,71 @@ internal sealed class SqlTriggerTargetScaler : ITargetScaler
     {
         private readonly SqlTriggerMetricsProvider _metricsProvider;
         private readonly int _maxChangesPerWorker;
+        private readonly ILogger _logger;
+        private readonly string _connectionString;
+        private readonly SqlObject _userTable;
+        private static DateTime _lastTableCheck = DateTime.MinValue;
+
 
         public SqlTriggerTargetScaler(string userFunctionId, SqlObject userTable, string userDefinedLeasesTableName, string connectionString, int maxChangesPerWorker, ILogger logger)
         {
             this._metricsProvider = new SqlTriggerMetricsProvider(connectionString, logger, userTable, userFunctionId, userDefinedLeasesTableName);
             this.TargetScalerDescriptor = new TargetScalerDescriptor(userFunctionId);
             this._maxChangesPerWorker = maxChangesPerWorker;
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._connectionString = !string.IsNullOrEmpty(connectionString) ? connectionString : throw new ArgumentNullException(nameof(connectionString));
+            this._userTable = userTable ?? throw new ArgumentNullException(nameof(userTable));
         }
 
         public TargetScalerDescriptor TargetScalerDescriptor { get; }
 
         public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
         {
-            SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+            try
+            {
+                SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+
+                // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
+                int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
 
-            // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
-            int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
+                return GetScaleResultInternal(concurrency, metrics.UnprocessedChangeCount);
+            }
+            catch (Exception ex)
+            {
+                if (ex is SqlException sqlEx && (sqlEx.Message.Contains("Invalid object name") ||
```

Comment:
1. Exception text may be localized, so we should never be relying on that
2. I think it would be far better to have the [T-SQL used to fetch the unprocessed changes ](https://github.com/Azure/azure-functions-sql-extension/blob/35ad03153a1e557d3a61f4a938b551ca4f083fb7/src/TriggerBinding/SqlTriggerMetricsProvider.cs#L100) work like this instead : 

```sql
IF GlobalStateTable NOT EXISTS OR LeasesTable NOT EXISTS
THEN
    SELECT COUNT_BIG(*) FROM CHANGETABLE(CHANGES {this._userTable.BracketQuotedFullName}, null)
ELSE
    -- the query as it is currently
END
```

Reply:
good catch! Thanks will look for the error code.